### PR TITLE
refactor(ctrl): migrate GitHub webhook onto pkg/webhook receiver

### PR DESCRIPTION
**Pull request by Flo4604**
Migrates ctrl-api's GitHub webhook from a hand-rolled http.Handler onto the `pkg/webhook` receiver.

- Transport chores (method policing, body-size cap, HMAC verification, routing, metrics, retry semantics) move to `pkg/webhook` + `verifiers/github`; the handler file shrinks to business logic only.
- `push` / `pull_request` keep their exact dispatch behavior: same payload parsing, same Restate ingress Send with the delivery id as idempotency key (now sourced from `event.ID` via `X-GitHub-Delivery`).
- Deliberate skips return `webhook.ErrIgnore` with a reason, so they 200 and surface as `outcome=ignored` in `unkey_webhook_events_total{provider="github"}`.
- GitHub webhooks get transport metrics for the first time.

Behavior edge cases: malformed payloads on verified requests now return 500 instead of 400 (GitHub does not auto-retry either way). Unknown event types use the receiver's ignored path.

Stacked on ENG-2865 (#6424).

## Test plan

```bash
mise exec -- bazel test //svc/ctrl/api/... //pkg/webhook/...
```

No Stripe or Tilt required. Existing webhook integration tests cover push dispatch, fork PRs, deleted branches, created branches, and invalid signature 401.

### CODE REVIEW

**Comment by Flo4604** [comment]
* docs(billing): compute billing and spend cap guides #6619
* feat(ctrl): Deploy spend-cap enforcement suspend/resume (ENG-2923) #6542
* feat(ctrl): rewire Deploy cancel through ctrl-api (ENG-2922) #6541
* feat(ctrl): Deploy teardown service (ENG-2922) #6540
* feat(billing): Deploy spend-cap detection and budget alerts (ENG-2904) #6486
* feat(dashboard): mirror Deploy included credit and local billing UI #6679
* dev: docker prune settings and scoped image watches for Tilt #6674
* feat(pkg): email sending package (Resend templates) #6456
* refactor(ctrl)!: per-workspace push invocations for deploy billing #6452
* test(billing): shadow-mode meter validation (ENG-2879) #6451
* feat(billing): meter active keys for Deploy #6450
* feat(clickhouse): source dimension for verifications #6448
* feat(dashboard): store Compute spend budget settings #6432
* **refactor(ctrl): migrate GitHub webhook onto pkg/webhook receiver #6427 👈**
* `main`

---
This stack of pull requests is managed by [jj-ryu](https://github.com/dmmulroy/jj-ryu).

File: svc/ctrl/api/github_webhook.go [hunk 1]
```diff
@@ -14,111 +12,53 @@ import (
...
-	}
-
-	logger.Info("GitHub webhook signature verified", "event", event)
-
-	deliveryID := r.Header.Get("X-GitHub-Delivery")
+// NewGitHubWebhook builds the /webhooks/github handler.
+func NewGitHubWebhook(restateClient *restateingress.Client, webhookSecret string) http.Handler {
+	s := &githubWebhook{restate: restateClient}
+	return webhook.New("github", githubverifier.New(webhookSecret)).
+		On(webhook.Typed(s.handlePush), "push").
```
Comment (chronark) [inline svc/ctrl/api/github_webhook.go]: this api reads weird, I get it's done to use variadic arguments, but what I read here is<br>"On s.handlePush do 'push'"<br><br>but it's actually the other way: "on push do s.handlePush"<br><br>I realise that's out of scope for this PR, but I wanted to raise it to maybe rethink this later.<br>I'd maybe do `On([]string{"push"], s.handlePush())`

**Review by chronark — COMMENTED** [review]
Why do we not return a proper 400? I get it that github doesn't care, but why don't we do the right thing anyways?

**Review by chronark — APPROVED** [review]

File: svc/ctrl/api/github_webhook.go [hunk 2]
```diff
@@ -14,111 +12,53 @@ import (
...
+	s := &githubWebhook{restate: restateClient}
+	return webhook.New("github", githubverifier.New(webhookSecret)).
+		On([]string{"push"}, webhook.Typed(s.handlePush)).
+		On([]string{"pull_request"}, webhook.Typed(s.handlePullRequest)).
+		// Branch lifecycle events carry no code to deploy; the first push of a
+		// new branch arrives as its own push event (created: true), which is
+		// what triggers preview deployments.
+		On([]string{"create", "delete", "installation"}, ignoreEvent).
+		// GitHub Apps receive every subscribed event type; anything without a
+		// handler is deliberately not deployment-relevant.
```
Comment (ogzhanolguncu) [inline svc/ctrl/api/github_webhook.go]: clean af

**Review by ogzhanolguncu — APPROVED** [review]
